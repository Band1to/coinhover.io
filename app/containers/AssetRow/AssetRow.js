import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

// Utils
import { percentClasser } from '../../utils/styler';
import { cleanNonNumeric, truncate } from '../../utils/formatter';
import {
  formatPercentage,
  numberWithCommas,
  rounder
} from '../../utils/math';

// Actions
import { updateCoin } from '../../actions';

export class AssetRow extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      asset: props.asset,
      balance: props.balance,
      value: rounder(props.balance, props.price_usd),
      percentage: props.percentage
    };

    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(event) {
    const balance = cleanNonNumeric(event.target.value);
    const priceUsd = this.state.asset.price_usd;
    const value = rounder(balance, priceUsd);
    const coin = this.state.asset;
    coin.value = value;
    this.props.dispatchUpdateCoin(coin);
    this.setState({ balance, value });
  }

  render() {
    const symbol = this.state.asset.symbol;
    const name = this.state.asset.name;
    const price = this.state.asset.price_usd;
    const priceUsd = `USD price ${this.state.asset.price_usd}`;
    const percentChange24h = this.state.asset.percent_change_24h;
    const percentage = formatPercentage(this.state.asset.percentage);
    const logo = this.state.asset.logo;

    return (
      <ul className="flex-container">
        <li className="flex-item flex-button">
          <div className="asset-btn">
            <div className="coin-logo">
              <img src={logo} alt={logo} />
            </div>
            <div className="asset-name">
              <strong>{ symbol }</strong>
              <small>{ truncate(name, 23) }</small>
            </div>
          </div>
        </li>
        <li className="flex-item num">
          <input
            type="text"
            className="white"
            value={this.state.balance}
            onChange={this.handleChange}
          />
        </li>
        <li className="flex-item num">
          <div className="flex-border" title={priceUsd}>
            ${price}
          </div>
        </li>
        <li className="flex-item num">
          <div className="flex-border" title={this.state.value}>
            ${numberWithCommas(this.state.value)}
          </div>
        </li>
        <li className={percentClasser(percentChange24h)}>
          <div className="flex-border">
            { this.state.asset.percentChange24h }%
          </div>
        </li>
        <li className="flex-item num">
          <div className="flex-border">{ percentage }%</div>
        </li>
        <li>
          <div className="close-x-smll" />
        </li>
      </ul>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  dispatchUpdateCoin(coin) {
    dispatch(updateCoin(coin));
  }
});

const AssetRowContainer = AssetRow;
export default connect(null, mapDispatchToProps)(AssetRowContainer);

AssetRow.propTypes = {
  price_usd: PropTypes.string.isRequired,
  value: PropTypes.string
};
