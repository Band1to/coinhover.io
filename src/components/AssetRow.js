import React from 'react'
import PropTypes from 'prop-types'
export default function AssetRow ({ asset }) {
	return (
		<ul className="flex-container">
			<li className="flex-item">
				<div className="asset-btn">
					<div className="icon-bitcoin"></div> {asset.name} ({asset.symbol})
				</div>
			</li>
			<li className="flex-item">
				<input type="text" placeholder="Holdings"/>
			</li>
			<li className="flex-item">{ asset.price_usd }</li>
			<li className="flex-item positive">{ asset.percent_change_1h }</li>
			<li className="flex-item">40%</li>
		</ul>
	)
}