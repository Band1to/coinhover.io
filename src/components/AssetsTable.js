import React from 'react'

class AssetsTable extends React.Component {
	render() {
		return (
			<section className="asset-table">

				<ul className="flex-container">
					<li className="flex-item">
						<div className="asset-btn">
							<div className="icon-bitcoin"></div> Bitcoin (BTC)
						</div>
					</li>
					<li className="flex-item">$2700</li>
					<li className="flex-item positive"><em>+</em>0.1</li>
					<li className="flex-item">40%</li>
				</ul>	

				<div className="add-asset-row">
					<div className="add-btn">
						<div className="icon-plus-outline"></div> Add Asset
					</div>
				</div>
			</section>
		)
	}
}

export default AssetsTable