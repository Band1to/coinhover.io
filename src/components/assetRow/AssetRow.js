import React from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import { updateCoin } from '../../reducer/portfolio/actions'
import { cleanNonNumeric, truncate } from '../../utils/formatter'
import { floorCents, formatPercentage, multiply, rounder } from '../../utils/math'
import { percentClasser } from '../../utils/styler'

const mapDispatchToProps = (dispatch) => ({
	dispatchUpdateCoin(coin) {
		dispatch(updateCoin(coin))
	}
});

export class AssetRow extends React.Component {

	constructor(props) {
		super(props)
		this.state = {
			asset: props.asset,
			balance: props.balance,
			value: rounder(props.balance, props.price_usd),
			percentage: props.percentage
		};

		this.handleChange = this.handleChange.bind(this);
	}

	handleChange(event) {
		const balance = cleanNonNumeric(event.target.value);
		const price_usd = this.state.asset.price_usd;
		const value = rounder(balance, price_usd);
		const coin = this.state.asset;
		coin.value = value;
		this.props.dispatchUpdateCoin(coin);
		this.setState({ balance, value });
	}

	render() {
		const symbol = this.state.asset.symbol;
		const name = this.state.asset.name;
		const price = this.state.asset.price_usd;
		const price_usd = `USD price ${this.state.asset.price_usd}`;
		const percent_change_24h = this.state.asset.percent_change_24h;
		const percentage = formatPercentage(this.state.asset.percentage);
		const logo = this.state.asset.logo;
		console.log('this.state.asset', this.state.asset);

		return (
			<ul className="flex-container">
				<li className="flex-item flex-button">
					<div className="asset-btn">
						<div className="coin-logo">
							<img src={ logo }/>
						</div>
						<div className="asset-name">
							<strong>{ symbol }</strong>
							<small>{ truncate(name, 23) }</small>
						</div>
					</div>
				</li>
				<li className="flex-item num">
					<input type="text"
						   className="white"
						   value={ this.state.balance }
						   onChange={ this.handleChange } />
				</li>
				<li className="flex-item num">
					<div className="flex-border" title={ price_usd }>
						${ price }
					</div>
				</li>
				<li className="flex-item num">
					<div className="flex-border" title={ this.state.value }>
						${ this.state.value }
					</div>
				</li>
				<li className={ percentClasser(percent_change_24h) }>
					<div className="flex-border">
						{ this.state.asset.percent_change_24h }%
					</div>
				</li>
				<li className="flex-item num">
					<div className="flex-border">{ percentage }%</div>
				</li>
			</ul>
		)
	}
}

const AssetRowContainer = AssetRow;
export default connect(null, mapDispatchToProps)(AssetRowContainer)

AssetRow.propTypes = {
	asset: PropTypes.object.isRequired,
	price_usd: PropTypes.string.isRequired
};