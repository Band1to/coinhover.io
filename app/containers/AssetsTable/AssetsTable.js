import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import AssetThead from '../../components/AssetTable/AssetThead';
import AssetRow from '../AssetRow/AssetRow';
import SearchCoin from '../Search/SearchCoin';

export class AssetsTable extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      search: false
    };
    this.closeSearch = this.closeSearch.bind(this);
  }

  closeSearch() {
    this.setState({ search: false });
  }

  render() {
    const assets = this.props.portfolio ? this.props.portfolio : [];

    const handleClick = (e) => {
	    e.preventDefault();
	    this.setState({ search: true });
    };

    return (
      <section className="asset-table">
        <AssetThead />
        { assets.map(asset =>
          (<AssetRow
            key={asset.id}
            asset={asset}
            balance={asset.balance}
            price_usd={asset.price_usd}
            percentage={asset.percentage}
          />)) }

        <div className="add-asset-row">
          { this.state.search ? (
            <SearchCoin closeSearch={this.closeSearch} />
          ) : (
            <div className="add-btn" onClick={handleClick}>
              + Add Asset
            </div>
          )}
        </div>
      </section>
    );
  }
}

const mapStateToProps = ({ portfolio, search }) => ({
  portfolio, search
});

const AssetsTableContainer = AssetsTable;
export default connect(mapStateToProps, null)(AssetsTableContainer);

AssetsTable.propTypes = {
  arrayWithShape: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired
  }))
};
