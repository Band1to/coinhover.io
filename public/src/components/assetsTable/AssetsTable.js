import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import AssetThead from './AssetThead'
import AssetRow from '../assetRow/AssetRow'
import SearchCoin from './SearchCoin'
import local_coins from '../../coins.json'
// import { coinsList } from '../../services/coinFactory'

const mapStateToProps = ({ portfolio }) => ({
	portfolio
});

export class AssetsTable extends React.Component {
	constructor(props) {
		super(props)
		this.state = {
			search: false
		};
		this.closeSearch = this.closeSearch.bind(this);
	}

	closeSearch() {
		this.setState({ search: false });
	}

	render() {
		const coins = local_coins;
		console.log('AssetsTable coins', coins)
		const assets = this.props.portfolio ? this.props.portfolio : [];

		const handleClick = (e) => {
	    e.preventDefault();
	    this.setState({ search: true });
		};

		return (
			<section className="asset-table">
				<AssetThead />
				{ assets.map(asset =>
					<AssetRow
						key={ asset.id }
						asset={ asset}
						balance={ asset.balance }
						price_usd={ asset.price_usd }
						percentage={ asset.percentage } />) }

				<div className="add-asset-row">
					{ this.state.search ? (
						<SearchCoin coins={ coins } closeSearch={ this.closeSearch } />
					) : (
						<div className="add-btn" onClick={ handleClick }>
							<div className="icon-plus-outline"></div> Add Asset
						</div>
					)}
				</div>
			</section>
		)
	}
}

const AssetsTableContainer = AssetsTable;
export default connect(mapStateToProps, null)(AssetsTableContainer)

AssetsTable.propTypes = {
  arrayWithShape: PropTypes.arrayOf(PropTypes.shape({
		id: PropTypes.string.isRequired,
		name: PropTypes.string.isRequired
  }))
}
