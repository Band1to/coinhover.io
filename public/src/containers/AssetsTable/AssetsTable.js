import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import AssetThead from '../../components/AssetTable/AssetThead'
import AssetRow from '../AssetRow/AssetRow'
import SearchCoin from '../Search/SearchCoin'
import local_coins from '../../coins.json'
// import { coinsList } from '../../services/coinFactory'

export class AssetsTable extends React.Component {
	constructor(props) {
		super(props)
		this.state = {
			search: false
		};
		this.closeSearch = this.closeSearch.bind(this);
	}

	closeSearch() {
		this.setState({ search: false });
	}

	render() {
		const coins = local_coins;
		const assets = this.props.portfolio ? this.props.portfolio : [];

		const handleClick = (e) => {
	    e.preventDefault();
	    this.setState({ search: true });
		};

		return (
			<section className="asset-table">
				<AssetThead />
				{ assets.map(asset =>
					<AssetRow
						key={ asset.id }
						asset={ asset}
						balance={ asset.balance }
						price_usd={ asset.price_usd }
						percentage={ asset.percentage } />) }

				<div className="add-asset-row">
					{ this.state.search ? (
						<SearchCoin coins={ coins } closeSearch={ this.closeSearch } />
					) : (
						<div className="add-btn" onClick={ handleClick }>
							<div className="icon-plus-outline"></div> Add Asset
						</div>
					)}
				</div>
			</section>
		)
	}
}

const mapStateToProps = ({ portfolio }) => ({
	portfolio
});

const AssetsTableContainer = AssetsTable;
export default connect(mapStateToProps, null)(AssetsTableContainer)

AssetsTable.propTypes = {
  arrayWithShape: PropTypes.arrayOf(PropTypes.shape({
		id: PropTypes.string.isRequired,
		name: PropTypes.string.isRequired
  }))
}
