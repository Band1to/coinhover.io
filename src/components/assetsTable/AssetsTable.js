import React from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import AssetThead from './AssetThead'
import AssetRow from '../assetRow/AssetRow'
import SearchCoin from './SearchCoin'

const mapStateToProps = ({ portfolio }) => ({
	portfolio
});

export class AssetsTable extends React.Component {
	constructor(props) {
		super(props)
		this.state = {
			search: false
		};
		this.closeSearch = this.closeSearch.bind(this);
	}

	closeSearch() {
		this.setState({ search: false });
	}

	render() {
		const assets = this.props.portfolio ? this.props.portfolio : [];
		console.log('AssetsTable assets', assets);

		const handleClick = (e) => {
		    e.preventDefault();
		    this.setState({ search: true });
		};

		return (
			<section className="asset-table">
				<AssetThead />
				{ assets.map(asset =>
					<AssetRow
						key={ asset.id }
						asset={ asset} 
						balance={ asset.balance }
						price_usd={ asset.price_usd }
						percentage={ asset.percentage } />) }
				
				<div className="add-asset-row">
					{ this.state.search ? (
						<SearchCoin closeSearch={ this.closeSearch } />
					) : (
						<div className="add-btn" onClick={ handleClick }>
							<div className="icon-plus-outline"></div> Add Asset
						</div>
					)}
				</div>
			</section>
		)
	}
}

const AssetsTableContainer = AssetsTable;
export default connect(mapStateToProps, null)(AssetsTableContainer)

AssetsTable.propTypes = {
   arrayWithShape: PropTypes.arrayOf(PropTypes.shape({
		id: PropTypes.string.isRequired,
		name: PropTypes.string.isRequired
   }))
}