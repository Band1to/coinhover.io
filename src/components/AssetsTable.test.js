import React from 'react'
import * as enzyme from 'enzyme';
import toJson from 'enzyme-to-json'
import AssetsTable from './AssetsTable'
import AssetRow from './AssetRow'

const assetsTable = enzyme.shallow(<AssetsTable />);

describe('<AssetsTable /> component', () => {
	it('should render', () => {
		const tree = toJson(assetsTable);
		expect(tree).toMatchSnapshot();
	});

	// it('contains the AssetsTable component', () => {
	// 	expect(assetsTable.find(AssetRow).length).toBe(1);
	// });
});