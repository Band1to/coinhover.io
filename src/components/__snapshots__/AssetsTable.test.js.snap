// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AssetsTable /> component should render 1`] = `
<section
  className="asset-table"
>
  <div
    className="add-asset-row"
  >
    <div
      className="add-btn"
    >
      <div
        className="icon-plus-outline"
      />
       Add Asset
    </div>
  </div>
</section>
`;
